code/datums/preferences.dm

```diff
@@ -159,8 +159,34 @@ datum/preferences
 		user << browse_rsc(icon(cursors_selection[target_cursor]), "tcursor.png")
 		user << browse_rsc(icon(hud_style_selection[hud_style], "preview"), "hud_preview.png")
 
-		var/dat = "<html><head><meta http-equiv=\"X-UA-Compatible\" content=\"IE=8\"/></head><body><title>Character Setup</title>"
-
+		var/dat = "<html><head><meta http-equiv=\"X-UA-Compatible\" content=\"IE=8\"/><meta http-equiv=\"pragma\" content=\"no-cache\" /></head><body><title>Character Setup</title>"
+		var/script = {"
+<script type='text/javascript' src='[resource("js/jquery.min.js")]'></script>
+<script type='text/javascript'>
+$(function() {
+	$('select').change(function(){
+        var id = document.activeElement.id
+		var r = $("#" + id + " option:selected" ).text();
+
+		window.location='byond://?src=\ref[src];preferences=1;id='+id+';style='+encodeURIComponent(r);
+		SwitchPic("sprite_preview");
+
+	});
+});
+
+
+function SwitchPic(picID) {

```

Comment:
Could move this function within the jquery dom ready block above (the `$(function() {`) to avoid polluting global scope.